Gems/Atom/RPI/Code/Tests/Material/MaterialTypeSourceDataTests.cpp

```diff
@@ -1952,4 +2022,213 @@ namespace UnitTest
 
         errorMessageFinder.CheckExpectedErrorsFound();
     }
+    
+    TEST_F(MaterialTypeSourceDataTests, LoadWithImportedJson)
+    {
+        const AZStd::string propertyGroupJson = R"(
+            {
+                "name": "myGroup",
+                "displayName": "My Group",
+                "description": "This group is defined in a separate JSON file",
+                "properties": [
+                    {
+                        "name": "foo",
+                        "type": "Bool"
+                    },
+                    {
+                        "name": "bar",
+                        "type": "Float"
+                    }
+                ]
+            }
+        )";
+
+        IO::FixedMaxPath propertyGroupJsonFilePath = m_tempFolder/"MyPropertyGroup.json";
+        AZ::Utils::WriteFile(propertyGroupJson, propertyGroupJsonFilePath.c_str());
```

Comment:
Yes, reading the referenced files from disk for the import command is an important part of the test. It might be possible to mock it in some way, but it's just as easy to do the real thing so might as well.
I think we have a lot of tests that write temp files to disk and then leave them lying around.